internal/clients/http/sources/sources_client.go

```diff
@@ -170,9 +174,12 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
 	defer span.End()
 
 	// Get all the authentications linked to a specific source
-	resp, err := c.client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &ListSourceAuthenticationsParams{}, headers.AddSourcesIdentityHeader, headers.AddEdgeRequestIdHeader)
+	resp, err := c.client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &ListSourceAuthenticationsParams{}, headers.AddSourcesIdentityHeader, headers.AddEdgeRequestIdHeader, func(ctx context.Context, req *stdhttp.Request) error {
+		req.URL.RawQuery = url.QueryEscape("filter[resource_type]") + "=Application" + "&" + url.QueryEscape("filter[authtype][starts_with]") + "=provisioning"
+		return nil
+	})
```

Comment:
Here you could use it as well.

If amount of arguments is not even number, it should panic.